odoo/addons/base/models/ir_actions_report.py

```diff
@@ -306,16 +306,15 @@ def _prepare_html(self, html):
 
         # Retrieve bodies
         for node in root.xpath(match_klass.format('article')):
-            body = layout.render(dict(subst=False, body=lxml.html.tostring(node), base_url=base_url))
-            bodies.append(body)
-            oemodelnode = node.find(".//*[@data-oe-model='%s']" % self.model)
-            if oemodelnode is not None:
-                res_id = oemodelnode.get('data-oe-id')
-                if res_id:
-                    res_id = int(res_id)
+            if node.get('data-oe-model') == self.model:
+                body = layout.render(dict(subst=False, body=lxml.html.tostring(node), base_url=base_url))
+                bodies.append(body)
+                res_ids.append(int(node.get('data-oe-id', False)))
             else:
-                res_id = False
-            res_ids.append(res_id)
+                raise UserError(_("The template does not contains data-oe-model on the 'div' tag with 'article' class"))
```

Comment:
add log with more informations & send to the user a message like 'contact your administrator'